phoenix-core/src/test/java/org/apache/phoenix/expression/InListExpressionTest.java

```diff
@@ -62,4 +70,68 @@ public void testHashCode() throws Exception {
                     + hashCode, firstHashCode, hashCode);
         }
     }
+
+    @Test
+    public void testGetSortedRowValueConstructorExpressionListNotPkOrder() {
+        testGetSortedRowValueConstructorExpressionList(false);
+    }
+
+    @Test
+    public void testGetSortedRowValueConstructorExpressionListPkOrder() {
+        testGetSortedRowValueConstructorExpressionList(true);
+    }
+
+    private void testGetSortedRowValueConstructorExpressionList(boolean isPkOrder) {
```

Comment:
Why don't we break this up more?  One we aren't testing the code in InListColumnKeyValuePair.  Two everything is being tested as part of the constructor.  I'd test it in 2 parts, 1 a InListRVCRewritte class/method that takes a array of indexes and a RVC and returns a new RVC with the new order.  2 a method that constructs an order array given a RVC of PKs.   Then its easy to test some of the scenarios like if we have less/more values than the array.

Reply:
@yanxinyi don't like that particular refactoring?  I think it will make the code a lot cleaner?

Reply:
It used to pass a list of expression that does sorting and reconstructing rvc.
the new implementation takes a list of `InListColumnKeyValuePair` instead so that I can test sorting logic and construction logic in a method and test reconstructed a new rvc in this method.